agent_build_refactored/managed_packages/managed_packages_builders.py

```diff
@@ -0,0 +1,932 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import hashlib
+import json
+import logging
+import shutil
+import subprocess
+import argparse
+import os
+import pathlib as pl
+from typing import List, Tuple, Optional, Dict
+
+
+from agent_build_refactored.tools.runner import Runner, RunnerStep, ArtifactRunnerStep, RunnerMappedPath, EnvironmentRunnerStep, DockerImageSpec,GitHubActionsSettings
+from agent_build_refactored.tools.constants import SOURCE_ROOT, DockerPlatform
+from agent_build_refactored.tools import check_call_with_log
+
+logger = logging.getLogger(__name__)
+
+"""
+This module is responsible for building Python agent for Linux distributions with package managers.
+
+It defines builder class that is responsible for the building of the Linux agent deb and rpm packages that are 
+    managed by package managers such as apt and yum.
+
+Besides building the agent package itself, it also builds dependency packages:
+    - scalyr-agent-python3
+    - scalyr-agent-libs
+
+These packages are needed to make agent package completely independent of a target system.
+
+
+The 'scalyr-agent-python3' package provides Python interpreter that is specially built to be used by the agent.
+    Its main feature that it is built against the oldest possible version of gLibc, so it has to be enough to maintain
+    only one build of the package in order to support all target systems.
+
+
+The 'scalyr-agent-libs' package provides requirement libraries for the agent, for example Python 'requests' or
+    'orjson' libraries. It is intended to ship it separately from the 'scalyr-agent-python3' because we update
+    agent's requirements much more often than Python interpreter.
+
+The structure of the mentioned packages has to guarantee that files of these packages does not interfere with
+    files of local system Python interpreter. To achieve that, the dependency packages files are installed in their
+    own 'sub-directories'
+
+    /usr/libe/scalyr-agent-2-dependencies
+    /usr/libexec/scalyr-agent-2-dependencies
+    /usr/shared/scalyr-agent-2-dependencies
+    /usr/include/scalyr-agent-2-dependencies
+
+    and agent from the 'scalyr-agent-2' package has to use the
+    '/usr/libexec/scalyr-agent-2-dependencies/scalyr-agent-2-python3' executable.
+"""
+
+# Name of the subdirectory of dependency packages.
+AGENT_DEPENDENCY_PACKAGE_SUBDIR_NAME = "scalyr-agent-2-dependencies"
+
+# Name of the dependency package with Python interpreter.
+PYTHON_PACKAGE_NAME = "scalyr-agent-python3"
+
+# name of the dependency package with agent requirement libraries.
+AGENT_LIBS_PACKAGE_NAME = "scalyr-agent-libs"
+
+
+EMBEDDED_PYTHON_VERSION = "3.11.0"
+EMBEDDED_PYTHON_SHORT_VERSION = ".".join(EMBEDDED_PYTHON_VERSION.split(".")[:2])
+
+PYTHON_PACKAGE_SSL_VERSION = "1.1.1k"
```

Comment:
Does this version here need to be the lowest common denominator for compatibility reasons? Or we can use new / latest openssl version? Also, I assume openssl is dynamically linked to OpenSSL?

Reply:
This has to be latest version. The plan is to add periodical job that checks crucial libraries such as OpenSSL for new versions/critical bug fixes.
 
Yes, It is dynamically linked and it is included to the Python package, so we can run the interpreter and linking to this version of OpenSSL

According to my tests, latest version of the OpenSSl does not have any compatibility issues with all distros that we plan to maintain. But it will be shown closer in one of the next PRs with AMI tests.